packages/provider-elasticsearch/src/utils/highlightUtil.test.js

```diff
@@ -0,0 +1,58 @@
+import { mergeHitHighlights } from './highlightUtil.js'
+
+describe('Highlight field aggregation', () => {
+  it('should combine subfield highlights with field highlights', () => {
+    let hitHighlights = {
+      'foo.bar': ['<b class="search-highlight">foo</b> this handle'],
+      'foo.bar.exact': ['foo this <b class="search-highlight">handle</b>'],
+      foo: ['foo this <b class="search-highlight">bar</b>'],
+      'foo.exact': ['<b class="search-highlight">foo</b> this bar'],
+      'foo.car.bar': ['<b class="search-highlight">foo</b> this is not merged'],
+    }
+    let fields = ['foo.bar', 'foo', 'foo.car.bar']
+
+    let nodeHighlight = {
+      pre_tags: ['<b class="search-highlight">'],
+      post_tags: ['</b>'],
+    }
+
+    expect(mergeHitHighlights(nodeHighlight, fields, hitHighlights)).toEqual({
+      'foo.bar': [
+        '<b class="search-highlight">foo</b> this <b class="search-highlight">handle</b>',
+      ],
+      foo: [
+        '<b class="search-highlight">foo</b> this <b class="search-highlight">bar</b>',
+      ],
+      'foo.car.bar': ['<b class="search-highlight">foo</b> this is not merged'],
+    })
+  })
+  it('should combine subfield highlights with field highlights1', () => {
```

Comment:
This function serves the same purpose as the one above. Also the field values are super noisy so it's difficult to detect where the highlighted substrings are.